main/src/ca/uwaterloo/flix/api/lsp/Visitor.scala

```diff
@@ -295,7 +295,7 @@ object Visitor {
   private def visitExpr(expr: Expr)(implicit a: Acceptor, c: Consumer): Unit = {
     if (!a.accept(expr.loc)) { return }
 
-    c.consumeExpr(expr)
+    if (!expr.loc.isSynthetic) c.consumeExpr(expr)
```

Comment:
This is not correct, generally speaking. It might work here, but it probably breaks other things that are desugared.

We need a more general and robust solution.

Also one that is applied to all nodes, not just exprs.

Reply:
Why is it not correct? isn't lsp about the stuff that the user has in their program?

Reply:
There are synthetic node with real nodes inside.

Reply:
right, but the synthetic one is still explored, just not consumed